File: openapi-generator/csharp-nancyfx/src/Org.OpenAPITools/Models/PrevJobStatus.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools.v1.Models
{
    /// <summary>
    /// The previous state of the dispatch job.
    /// </summary>
    /// <value>The previous state of the dispatch job.</value>
    public enum PrevJobStatus
    {
        
        /// <summary>
        /// Enum Unassigned
        /// </summary>
        Unassigned = 1,
        
        /// <summary>
        /// Enum Scheduled
        /// </summary>
        Scheduled = 2,
        
        /// <summary>
        /// Enum EnRoute
        /// </summary>
        EnRoute = 3,
        
        /// <summary>
        /// Enum Arrived
        /// </summary>
        Arrived = 4,
        
        /// <summary>
        /// Enum Completed
        /// </summary>
        Completed = 5,
        
        /// <summary>
        /// Enum Skipped
        /// </summary>
        Skipped = 6
    }

}

File: openapi-generator/java-play-framework/app/apimodels/PrevJobStatus.java
package apimodels;

import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * The previous state of the dispatch job.
 */
public enum PrevJobStatus {
  
  UNASSIGNED("JobState_Unassigned"),
  
  SCHEDULED("JobState_Scheduled"),
  
  ENROUTE("JobState_EnRoute"),
  
  ARRIVED("JobState_Arrived"),
  
  COMPLETED("JobState_Completed"),
  
  SKIPPED("JobState_Skipped");

  private final String value;

  PrevJobStatus(String value) {
    this.value = value;
  }

  @Override
  @JsonValue
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static PrevJobStatus fromValue(String text) {
    for (PrevJobStatus b : PrevJobStatus.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + text + "'");
  }
}

